File: sdk/nodejs/audit-sdk.js
const axios = require('axios');
const { v4: uuidv4 } = require('uuid');

class AuditSDK {
  constructor(ingestionUrl, serviceName) {
    this.ingestionUrl = ingestionUrl;
    this.serviceName = serviceName;
  }

  async emit(action, userId, resourceType, resourceId, metadata = {}) {
    const event = {
      id: uuidv4(),
      action,
      user_id: userId,
      resource_type: resourceType,
      resource_id: resourceId,
      timestamp: new Date().toISOString(),
      schema_version: "v1",
      metadata
    };

    try {
      await axios.post(this.ingestionUrl, event);
    } catch (error) {
      console.error("[AuditSDK] Failed to emit event:", error.message);
    }
  }
}

module.exports = AuditSDK;

File: sdk/python/audit_sdk.py
import requests
import uuid
import time

class AuditSDK:
    def __init__(self, ingestion_url, service_name):
        self.ingestion_url = ingestion_url
        self.service_name = service_name

    def emit(self, action, user_id, resource_type, resource_id, metadata=None):
        event = {
            "id": str(uuid.uuid4()),
            "action": action,
            "user_id": user_id,
            "resource_type": resource_type,
            "resource_id": resource_id,
            "timestamp": time.time(),
            "schema_version": "v1",
            "metadata": metadata or {}
        }
        try:
            requests.post(self.ingestion_url, json=event, timeout=3)
        except Exception as e:
            print(f"[AuditSDK] Failed to emit event: {e}")
